test/functional/deprecated/commandsFunctional.js

```diff
@@ -16,76 +16,69 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 /* eslint-env mocha */
+/* eslint-disable new-cap */
+
 
 const { expect } = require('chai');
 const { parseString } = require('xml2js');
 const {
-  iCmd, iSh, iQsh,
+  iCmd, iSh, iQsh, iConn,
 } = require('../../../lib/itoolkit');
 
-const { returnTransportsDeprecated } = require('../../../lib/utils');
+const { config } = require('../config');
 
-// Set Env variables or set values here.
-const opt = {
-  database: process.env.TKDB || '*LOCAL',
-  username: process.env.TKUSER || '',
-  password: process.env.TKPASS || '',
-  host: process.env.TKHOST || 'localhost',
-  port: process.env.TKPORT || 80,
-  path: process.env.TKPATH || '/cgi-bin/xmlcgi.pgm',
-};
+if (config.transport !== 'idb' && config.transport !== 'rest') {
+  throw new Error('Only idb and rest transports are available for deprecated tests');
```

Comment:
Why are idb and rest the only supported transports?

Reply:
The functional/deprecated/*.js  test create a connection object using deprecated `iConn` class. This class only supported `rest` and `idb` transports.

The purpose of these tests is to test compatibility using the old way.

Reply:
Ok, makes sense. We should probably skip them instead in that case.

Reply:
We can skip tests prgoramatically with:

![image](https://user-images.githubusercontent.com/33973272/77574985-d4121900-6ea0-11ea-9e3a-5b5c4033be14.png)

https://mochajs.org/#inclusive-tests

With this we can move tests from `tests/functional/deprecated` folder into their respective test file  in `test/functional`. Just need to add a describe block for the deprecated tests and test if idb or rest transport is being used used. If not we can just skip those deprecated tests.

Reply:
Apparently this doesn't work with arrow functions: https://mochajs.org/#arrow-functions Mocha recommends not using arrow functions for this reason.

Reply:
Yes we wont be able to use `this.skip()` until #200 is resolved. Should we leave it as is for now and fix up later during 1.1 milestone?

Reply:
Yeah, we can fix this up later I think.

Reply:
Alternatively, we can have the test do nothing in that case and put a comment saying to do `this.skip()` instead once #200 is resolved.